plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -15,39 +34,70 @@
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Link;
 import org.eclipse.swt.widgets.Text;
-import org.eclipse.ui.PartInitException;
-import org.eclipse.ui.PlatformUI;
-import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
 import org.eclipse.ui.dialogs.PropertyPage;
 import org.eclipse.ui.forms.widgets.ExpandableComposite;
+import org.eclipse.ui.preferences.ScopedPreferenceStore;
 
 import com.google.cloud.tools.eclipse.ui.util.FontUtil;
+import com.google.cloud.tools.eclipse.ui.util.databinding.BucketNameValidator;
+import com.google.cloud.tools.eclipse.ui.util.databinding.ProjectIdValidator;
+import com.google.cloud.tools.eclipse.ui.util.databinding.ProjectVersionValidator;
+import com.google.cloud.tools.eclipse.ui.util.event.OpenUrlSelectionListener;
+import com.google.cloud.tools.eclipse.util.AdapterUtil;
+import com.google.common.base.Preconditions;
 
 public class DeployPropertyPage extends PropertyPage {
 
   private static final String APPENGINE_DASHBOARD_URL = "https://console.cloud.google.com/appengine";
+
   private static final int INDENT_CHECKBOX_ENABLED_WIDGET = 10;
 
-  private Button promptForProjectIdButton;
+  private static final String PREFERENCE_STORE_DEFAULTS_QUALIFIER =
+      "com.google.cloud.tools.eclipse.appengine.deploy.defaults";
+  private static final String PREFERENCE_STORE_QUALIFIER = "com.google.cloud.tools.eclipse.appengine.deploy";
+  private static final String PREF_PROMPT_FOR_PROJECT_ID = "project.id.promptOnDeploy"; // boolean
+  private static final String PREF_PROJECT_ID = "project.id";
+  private static final String PREF_OVERRIDE_DEFAULT_VERSIONING = "project.version.overrideDefault"; // boolean
+  private static final String PREF_CUSTOM_VERSION = "project.version";
+  private static final String PREF_ENABLE_AUTO_PROMOTE = "project.promote"; // boolean
+  private static final String PREF_OVERRIDE_DEFAULT_BUCKET = "project.bucket.overrideDefault"; // boolean
+  private static final String PREF_CUSTOM_BUCKET = "project.bucket";
+
+  private static Logger logger = Logger.getLogger(DeployPropertyPage.class.getName());
+
   private Label projectIdLabel;
   private Text projectId;
-  private Button browseProjectButton;
-  
+  private Button promptForProjectIdButton;
+
   private Button overrideDefaultVersionButton;
   private Label versionLabel;
   private Text version;
-  
+
   private Button autoPromoteButton;
-  
+
   private Button overrideDefaultBucketButton;
   private Label bucketLabel;
   private Text bucket;
-  private Button browseBucketButton;
-  
+
+  private Model model = new Model();
+  private ObservablesManager observables;
+  private DataBindingContext bindingContext;
+
+
+  public DeployPropertyPage() {
+    DefaultScope.INSTANCE.getNode(PREFERENCE_STORE_DEFAULTS_QUALIFIER).putBoolean(PREF_PROMPT_FOR_PROJECT_ID, true);
+  }
```

Comment:
Sorry, I wasn't clear.  Currently you're configuring defaults in the `Model` initialization, except for this part.  But a user can start a deploy without ever opening this properties page, and so no defaults will have been written out to the project preferences.  So the deploy code won't see any pref values.  If we instead set the defaults using a preference initializer (via the `org.eclipse.equinox.preferences.preferences` extension point), then providing the deploy code uses `IPreferenceService` (either via `Platform.getPreferenceService()` or via the OSGi service) or `ScopedPreferenceStore`, which lookup any unresolved preferences in the default scope, it will see the 'right' defaults.  Then we can set the defaults for `Model` via `loadPreferences()` — and restore the 'restore defaults` button too.